languages.toml

```diff
@@ -537,3 +537,12 @@ scope = "source.regex"
 injection-regex = "regex"
 file-types = ["regex"]
 roots = []
+
+[[language]]
+name = "git-config"
+scope = "source.gitconfig"
+roots = []
+file-types = [".gitmodules", ".gitconfig"]
```

Comment:
Is this a good idea? It could be `.git/config` as well.

Reply:
Hmm yeah I think if file-types could be a regex or a glob that'd work? We'd want to detect git-config for a few files where the containing directory/ies are important

Reply:
For git it's actually a little more complicated. Git repositories don't have to be called .git. That is just the default for working trees. There many other possible variations. e.g bare repositories, submodules and git with multiple working trees (see git-worktree).

`.git/modules/helix-syntax/languages/tree-sitter-c/config` for example is probably a git config file in your helix working tree.

The best way afaik to detect git config files is to check for other items in the same directory. objects and refs for example.

I'm pretty sure other file types will be in a similar situation.

bat seems to have this figured out. Perhaps would be worth looking to see how they do it?

Reply:
I am thinking that's better done as plugin or some sort of check then.